### DISCUSSION TITLE
Cloudflare优选IP - Busyo's Blog

### DISCUSSION THREAD

**Discussion by busyoGG** [busyoGG/busyoGG.github.io#64] [Announcements]
https://busyo.buzz/article/8583c53078ec/ 

 利用优选IP加快Cloudflare的访问速度，防止因为IP被墙导致的无法访问或者访问缓慢 - Busyo - Busyo's Blog

**Comment by KKKKKKD32** [comment]
报错
Traceback (most recent call last):
  File "C:\Users\K\Desktop\cloudf\11.py", line 81, in <module>
    url = url.replace('{RECORD_ID}', DNS_ID)
                                     ^^^^^^
NameError: name 'DNS_ID' is not defined

**Comment by busyoGG** [comment]
> @KKKKKKD32
> 报错
> Traceback (most recent call last):
>   File "C:\Users\K\Desktop\cloudf\11.py", line 81, in <module>
>     url = url.replace('{RECORD_ID}', DNS_ID)
>                                      ^^^^^^
> NameError: name 'DNS_ID' is not defined

我自己是正常的，可能是因为DNS_ID没有获取到值

**Comment by KKKKKKD32** [comment]
有什么办法解决吗

**Comment by busyoGG** [comment]
> @KKKKKKD32
> 有什么办法解决吗
> 

我是没啥办法，只能靠自己慢慢排查了，有问题问问AI（

**Comment by KKKKKKD32** [comment]
if resJson['success']:
    data = resJson['result']
    DNS_ID = None
    for record in data:
        print("检查记录:", record)  # 输出调试信息
        if record['name'] == RECORD_NAME and record['type'] == 'A':
            DNS_ID = record['id']
            print("A 记录名称为", RECORD_NAME, "的 ID 为：", DNS_ID)
好像是这地方的问题，DNS_ID的值被设置为没有了

**Comment by busyoGG** [comment]
> @KKKKKKD32
> if resJson['success']:
>     data = resJson['result']
>     DNS_ID = None
>     for record in data:
>         print("检查记录:", record)  # 输出调试信息
>         if record['name'] == RECORD_NAME and record['type'] == 'A':
>             DNS_ID = record['id']
>             print("A 记录名称为", RECORD_NAME, "的 ID 为：", DNS_ID)
> 好像是这地方的问题，DNS_ID的值被设置为没有了

主要问题就是为什么DNS_ID的值不存在，我自己运行是正常获取的，你可能要排查这个问题

**Comment by 4jigo** [comment]
NameError                                 Traceback (most recent call last)
<ipython-input-6-5bc6b0a01d83> in <cell line: 81>()
     79 url = 'https://api.cloudflare.com/client/v4/zones/{ZONE_ID}/dns_records/{RECORD_ID}'
     80 url = url.replace('{ZONE_ID}', ZONE_ID)
---> 81 url = url.replace('{RECORD_ID}', DNS_ID)
     82 # 发送 API 请求
     83 response = requests.request("PUT",url, json=data, headers=headers)

NameError: name 'DNS_ID' is not defined
